Assets/Scripts/FlashLight.cs

```diff
@@ -52,25 +61,35 @@ private void LateUpdate()
         {
             AutoAim();
         }
-
     }
+
     private void TurnOnOffLight()
     {
         if (input.FlashlightDown)
         {
-            if (!isOn && Charge > 0f)
+            if (Charge > 0f)
             {
-                lightSource.enabled = true;
-                anim.SetBool("IsOn", true);
-                isOn = true;
+                if (!isOn)
+                {
+                    lightSource.enabled = true;
+                    anim.SetBool("IsOn", true);
+                    isOn = true;
+                    PlaySound();
+                }
+                else
+                {
+                    lightSource.enabled = false;
+                    anim.SetBool("IsOn", false);
+                    isOn = false;
+                    PlaySound();
```

Comment:
no need for the if statement here. just do the following:
                    isOn = !isOn;
                    lightSource.enabled = isOn;
                    anim.SetBool("IsOn", isOn);
                    PlaySound();